### PR TITLE
Refactor IOS libs to use cliconf

### PR DESCRIPTION
**Pull request by rcarrillocruz**

### CODE REVIEW

**Comment by rcarrillocruz** [comment]
bot_skip

**Review by ganeshrn — APPROVED** [review]
lgtm.

**Comment by ganeshrn** [comment]
Please resolve merge conflict before merging.

File: lib/ansible/plugins/cliconf/ios.py [hunk 1]
```diff
@@ -64,11 +64,22 @@ def get_config(self, source='running'):
 
     @enable_mode
     def edit_config(self, command):
-        for cmd in chain(['configure terminal'], to_list(command), ['end']):
-            self.send_command(to_bytes(cmd))
+        for cmd in chain([b'configure terminal'], to_list(command), [b'end']):
```
Comment (Qalthos) [inline lib/ansible/plugins/cliconf/ios.py]: This shouldn't be byte strings (they cast to bytes later anyway), and byte strings will break `json.loads()` in Python 3

**Review by Qalthos — CHANGES_REQUESTED** [review]

**Comment by rcarrillocruz** [comment]
merging, failures are CI issues unrelated to change

**Pull request merged**